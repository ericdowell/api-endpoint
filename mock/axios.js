export function BasicMock(options) {
  return new Promise((resolve, reject) => {
    if (options.errors) {
      return reject({
        config: {},
        response: {
          data: {
            errors: options.errors
          },
          status: options.status || 422
        }
      })
    }
    return resolve({
      config: {},
      data: { foo: 'bar' }
    })
  })
}
