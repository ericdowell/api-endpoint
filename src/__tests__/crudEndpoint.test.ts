import axios from 'axios'
import { CrudEndpoint } from '../index'
import { BasicMock } from './mock/axios'

jest.spyOn(axios, 'request').mockImplementation(BasicMock)

const callEndpoint = (instance: CrudEndpoint): {[key: string]: Function} => {
  return {
    delete: instance.delete.bind(instance),
    get: instance.get.bind(instance),
    patch: instance.patch.bind(instance),
    post: instance.post.bind(instance),
    put: instance.put.bind(instance),
  }
}

describe(`${CrudEndpoint.name}`, (): void => {
  it.each([
    ['delete'],
    ['get'],
    ['patch'],
    ['post'],
    ['put'],
  ])('the "%s" method calls request on parent', async(method): Promise<void> => {
    expect.assertions(1)
    const instance = new CrudEndpoint()
    const request = jest.spyOn(instance, 'request').mockImplementation(async(): Promise<any> => true)
    await callEndpoint(instance)[method]('path', { url: 'value' })
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(request.mock.calls[0]).toMatchSnapshot()
  })
})
