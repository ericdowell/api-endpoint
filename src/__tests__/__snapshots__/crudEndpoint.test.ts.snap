// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrudEndpoint the "delete" method calls request on parent 1`] = `
Array [
  "path",
  "delete",
  Object {
    "url": "value",
  },
]
`;

exports[`CrudEndpoint the "get" method calls request on parent 1`] = `
Array [
  "path",
  "get",
  Object {
    "url": "value",
  },
]
`;

exports[`CrudEndpoint the "patch" method calls request on parent 1`] = `
Array [
  "path",
  "patch",
  Object {
    "url": "value",
  },
]
`;

exports[`CrudEndpoint the "post" method calls request on parent 1`] = `
Array [
  "path",
  "post",
  Object {
    "url": "value",
  },
]
`;

exports[`CrudEndpoint the "put" method calls request on parent 1`] = `
Array [
  "path",
  "put",
  Object {
    "url": "value",
  },
]
`;
